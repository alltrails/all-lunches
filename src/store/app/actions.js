import { createAsyncAction, createAsyncTypes } from 'lib/reduxUtils';
// Async
export const INITIALIZE_APPLICATION = createAsyncTypes('app/initializeApplication');
export const initializeApplication = createAsyncAction(INITIALIZE_APPLICATION);
