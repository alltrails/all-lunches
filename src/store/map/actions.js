import { createAsyncAction, createAsyncTypes } from 'lib/reduxUtils';

// Async
export const ADD_REPORT = createAsyncTypes('map/addReport');
export const addReport = createAsyncAction(ADD_REPORT);

export const QUERY_AREA = createAsyncTypes('map/queryArea');
export const queryArea = createAsyncAction(QUERY_AREA);
