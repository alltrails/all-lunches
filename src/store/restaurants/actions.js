import { createAction, createAsyncAction, createAsyncTypes } from 'lib/reduxUtils';

// Sync
export const SET_HIGHLIGHTED_RESTAURANT_ID = 'restaurants/setHighlightedRestaurantId';
export const setHighlightedRestaurantId = createAction(SET_HIGHLIGHTED_RESTAURANT_ID);

export const SET_FILTER_OPTION = 'restaurants/setFilterOption';
export const setFilterOption = createAction(SET_FILTER_OPTION);

// Async
export const QUERY_AREA = createAsyncTypes('restaurants/queryArea');
export const queryArea = createAsyncAction(QUERY_AREA);

export const UPDATE_FAVORITED_RESTAURANTS = createAsyncTypes(
  'restaurants/updateFavoritedRestaurants',
);
export const updateFavoritedRestaurants = createAsyncAction(UPDATE_FAVORITED_RESTAURANTS);

export const FETCH_FAVORITED_RESTAURANTS = createAsyncTypes(
  'restaurants/fetchFavoritedRestaurants',
);
export const fetchFavoritedRestaurants = createAsyncAction(FETCH_FAVORITED_RESTAURANTS);
