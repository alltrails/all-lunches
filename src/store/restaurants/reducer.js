/* eslint-disable default-param-last */
import {
  UPDATE_FAVORITED_RESTAURANTS,
  FETCH_FAVORITED_RESTAURANTS,
  QUERY_AREA,
  SET_FILTER_OPTION,
  SET_SELECTED_RESTAURANT_ID,
} from './actions';

export const INITIAL_STATE = {
  favoritedItemIds: [],
  highlightedRestaurantId: null,
  isLoading: false,
  isUpdatingFavorites: false,
  restaurants: null,
  selectedFilterOption: null,
};

export default (state = INITIAL_STATE, action) => {
  const { payload, type } = action;

  switch (type) {
    case SET_FILTER_OPTION:
      return {
        ...state,
        selectedFilterOption: payload,
      };
    case SET_SELECTED_RESTAURANT_ID:
      return {
        ...state,
        highlightedRestaurantId: payload,
      };
    case UPDATE_FAVORITED_RESTAURANTS.PENDING:
      return {
        ...state,
        isUpdatingFavorites: true,
      };
    case UPDATE_FAVORITED_RESTAURANTS.ERROR:
      return {
        ...state,
        isUpdatingFavorites: false,
      };
    case UPDATE_FAVORITED_RESTAURANTS.SUCCESS:
      return {
        ...state,
        favoritedItemIds: payload,
        isUpdatingFavorites: false,
      };
    case FETCH_FAVORITED_RESTAURANTS.SUCCESS:
      return {
        ...state,
        favoritedItemIds: payload,
      };
    case QUERY_AREA.PENDING:
      return {
        ...state,
        isLoading: true,
      };
    case QUERY_AREA.ERROR:
      return {
        ...state,
        isLoading: false,
      };
    case QUERY_AREA.SUCCESS:
      return {
        ...state,
        isLoading: false,
        restaurants: payload,
      };
    default:
      return state;
  }
};
