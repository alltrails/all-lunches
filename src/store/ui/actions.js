import { createAction } from 'lib/reduxUtils';

// Sync
export const TOGGLE_OPEN_TOAST = 'ui/toggleOpenToast';
export const toggleOpenToast = createAction(TOGGLE_OPEN_TOAST);
