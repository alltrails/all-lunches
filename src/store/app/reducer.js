/* eslint-disable @typescript-eslint/default-param-last */
import { INITIALIZE_APPLICATION } from './actions';

export const INITIAL_STATE = {
  error: null,
  isLoading: false,
};

export default (state = INITIAL_STATE, action) => {
  const { payload, type } = action;

  switch (type) {
    case INITIALIZE_APPLICATION.PENDING:
      return {
        ...state,
        error: null,
        isLoading: true,
      };
    case INITIALIZE_APPLICATION.ERROR:
      return {
        ...state,
        error: payload,
        isLoading: false,
      };
    case INITIALIZE_APPLICATION.SUCCESS:
      return {
        ...state,
        isLoading: false,
      };
    default:
      return state;
  }
};
