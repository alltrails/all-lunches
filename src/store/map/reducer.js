/* eslint-disable default-param-last */
import { mergeReports } from 'lib/mapUtils';
import { ADD_REPORT, QUERY_AREA } from './actions';

export const INITIAL_STATE = {
  isLoading: false,
  reports: null,
  numOfReports: 0,
};

export default (state = INITIAL_STATE, action) => {
  const { payload, type } = action;

  switch (type) {
    case ADD_REPORT.PENDING:
      return {
        ...state,
        isLoading: true,
      };
    case ADD_REPORT.ERROR:
      return {
        ...state,
        isLoading: false,
      };
    case ADD_REPORT.SUCCESS:
      return {
        ...state,
        isLoading: false,
        reports: state.reports ? mergeReports(state.reports, payload) : payload,
        numOfReports: state.numOfReports ? state.numOfReports + 1 : 1,
      };
    case QUERY_AREA.SUCCESS: {
      const { numOfReports, reports } = payload;

      return {
        ...state,
        reports,
        numOfReports,
      };
    }

    default:
      return state;
  }
};
