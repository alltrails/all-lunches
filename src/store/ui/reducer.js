/* eslint-disable default-param-last */

import { TOGGLE_OPEN_TOAST } from 'store/ui/actions';
import { QUERY_AREA, UPDATE_FAVORITED_RESTAURANTS } from 'store/restaurants/actions';

export const INITIAL_STATE = {
  currentOpenToast: null,
};

const successToastMessages = {
  [UPDATE_FAVORITED_RESTAURANTS.SUCCESS]: {
    message: 'Successfully added your favorite restaurant!',
    type: 'success',
  },
  [QUERY_AREA.SUCCESS]: { message: 'Successfully fetched restaurants!', type: 'success' },
};

export default (state = INITIAL_STATE, action) => {
  const { payload, type } = action;

  switch (type) {
    case TOGGLE_OPEN_TOAST:
      return {
        ...state,
        currentOpenToast: payload,
      };
    case UPDATE_FAVORITED_RESTAURANTS.PENDING:
    case QUERY_AREA.PENDING:
      return {
        ...state,
        currentOpenToast: null,
      };
    case UPDATE_FAVORITED_RESTAURANTS.SUCCESS:
    case QUERY_AREA.SUCCESS:
      return {
        ...state,
        currentOpenToast: successToastMessages[type],
      };
    case UPDATE_FAVORITED_RESTAURANTS.ERROR:
    case QUERY_AREA.ERROR:
      return {
        ...state,
        currentOpenToast: { message: payload, type: 'error' },
      };
    default:
      return state;
  }
};
