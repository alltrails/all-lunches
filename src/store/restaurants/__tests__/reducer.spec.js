/* eslint-env jest */
import { restaurantsMock } from '__mocks__';
import * as types from '../actions';
import reducer, { INITIAL_STATE } from '../reducer';

describe('Restaurants reducer', () => {
  const isLoadingState = {
    ...INITIAL_STATE,
    isLoading: true,
  };

  const isUpdatingFavoritesState = {
    ...INITIAL_STATE,
    isUpdatingFavorites: true,
  };

  it('returns the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  describe('SET_FILTER_OPTION', () => {
    it('works as expected', () => {
      const testPayload = 'high_to_low';

      expect(
        reducer(INITIAL_STATE, {
          payload: testPayload,
          type: types.SET_FILTER_OPTION,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('SET_SELECTED_RESTAURANT_ID', () => {
    it('works as expected', () => {
      const testPayload = 'some_id';

      expect(
        reducer(INITIAL_STATE, {
          payload: testPayload,
          type: types.SET_SELECTED_RESTAURANT_ID,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('UPDATE_FAVORITED_RESTAURANTS.PENDING', () => {
    it('works as expected', () => {
      expect(
        reducer(INITIAL_STATE, {
          type: types.UPDATE_FAVORITED_RESTAURANTS.PENDING,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('UPDATE_FAVORITED_RESTAURANTS.SUCCESS', () => {
    it('works as expected', () => {
      const testPayload = ['some_id_1', 'some_id_2'];

      expect(
        reducer(isUpdatingFavoritesState, {
          payload: testPayload,
          type: types.UPDATE_FAVORITED_RESTAURANTS.SUCCESS,
        }),
      ).toMatchSnapshot();
    });

    it('works as expected when empty array', () => {
      const testPayload = [];

      expect(
        reducer(isUpdatingFavoritesState, {
          payload: testPayload,
          type: types.UPDATE_FAVORITED_RESTAURANTS.SUCCESS,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('UPDATE_FAVORITED_RESTAURANTS.ERROR', () => {
    it('works as expected', () => {
      expect(
        reducer(isLoadingState, {
          payload: 'looks like there was an error',
          type: types.UPDATE_FAVORITED_RESTAURANTS.ERROR,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('FETCH_FAVORITED_RESTAURANTS.SUCCESS', () => {
    it('works as expected', () => {
      const testPayload = ['some_id_1', 'some_id_2'];

      expect(
        reducer(isUpdatingFavoritesState, {
          payload: testPayload,
          type: types.FETCH_FAVORITED_RESTAURANTS.SUCCESS,
        }),
      ).toMatchSnapshot();
    });

    it('works as expected when empty array', () => {
      const testPayload = [];

      expect(
        reducer(isUpdatingFavoritesState, {
          payload: testPayload,
          type: types.FETCH_FAVORITED_RESTAURANTS.SUCCESS,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('QUERY_AREA.PENDING', () => {
    it('works as expected', () => {
      expect(
        reducer(INITIAL_STATE, {
          type: types.QUERY_AREA.PENDING,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('QUERY_AREA.SUCCESS', () => {
    it('works as expected', () => {
      const testPayload = restaurantsMock;

      expect(
        reducer(isLoadingState, {
          payload: testPayload,
          type: types.QUERY_AREA.SUCCESS,
        }),
      ).toMatchSnapshot();
    });
  });

  describe('QUERY_AREA.ERROR', () => {
    it('works as expected', () => {
      expect(
        reducer(isLoadingState, {
          payload: 'looks like there was an error',
          type: types.QUERY_AREA.ERROR,
        }),
      ).toMatchSnapshot();
    });
  });
});
