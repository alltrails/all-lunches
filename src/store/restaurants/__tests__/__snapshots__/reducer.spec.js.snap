// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurants reducer FETCH_FAVORITED_RESTAURANTS.SUCCESS works as expected 1`] = `
Object {
  "favoritedItemIds": Array [
    "some_id_1",
    "some_id_2",
  ],
  "highlightedRestaurantId": null,
  "isLoading": false,
  "isUpdatingFavorites": true,
  "restaurants": null,
  "selectedFilterOption": null,
}
`;

exports[`Restaurants reducer FETCH_FAVORITED_RESTAURANTS.SUCCESS works as expected when empty array 1`] = `
Object {
  "favoritedItemIds": Array [],
  "highlightedRestaurantId": null,
  "isLoading": false,
  "isUpdatingFavorites": true,
  "restaurants": null,
  "selectedFilterOption": null,
}
`;

exports[`Restaurants reducer QUERY_AREA.ERROR works as expected 1`] = `
Object {
  "favoritedItemIds": Array [],
  "highlightedRestaurantId": null,
  "isLoading": false,
  "isUpdatingFavorites": false,
  "restaurants": null,
  "selectedFilterOption": null,
}
`;

exports[`Restaurants reducer QUERY_AREA.PENDING works as expected 1`] = `
Object {
  "favoritedItemIds": Array [],
  "highlightedRestaurantId": null,
  "isLoading": true,
  "isUpdatingFavorites": false,
  "restaurants": null,
  "selectedFilterOption": null,
}
`;

exports[`Restaurants reducer QUERY_AREA.SUCCESS works as expected 1`] = `
Object {
  "favoritedItemIds": Array [],
  "highlightedRestaurantId": null,
  "isLoading": false,
  "isUpdatingFavorites": false,
  "restaurants": Array [
    Object {
      "coordinates": Array [
        -122.408999,
        37.7902566,
      ],
      "id": "ChIJkVLoONZ_j4ARx2vJ4umERnI",
      "name": "Uncle Vito's Pizza",
      "photoUrl": "https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEBdIFbWc8LjoO8hOGn1QTJBeq21XOCk6ZHFGBmGWWqrImCdbERWb-K4T9Mq_QB5Mspi1Kiyc_GTU_XRbcReDLyGaIWdJWcssnsj3mZmy5xQsGCPeuOiBjjDEciT92r5iUeEAvSNn4dPN2r9rng9SH3fTXpxlT6BA5TB1MPzYfQ_GU5J&3u4032&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=10623",
      "placeUrl": "https://www.google.com/maps/search/?api=1&query=Uncle%20Vito's%20Pizza&query_place_id=ChIJkVLoONZ_j4ARx2vJ4umERnI",
      "priceLevel": 1,
      "rating": 4.3,
      "supportingText": "restaurant",
      "userRatingsTotal": 1427,
    },
    Object {
      "coordinates": Array [
        -122.408675,
        37.7900735,
      ],
      "id": "ChIJg7cNIYyAhYAR8D4ml1oItTg",
      "name": "Roxanne's Cafe",
      "photoUrl": "https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEB-N4JLe8ZfukWW8G8Jum72Mw6dOey3xmpS8ywrT9kWjlPfrwgcjLFYrlY1NKGCMEDPTW0I4fgrbAmcVyf5ZEsE49LXdhibUiM-E4EKVyJQ2_wmj_NRZ_i999eUks-hnCQQZAeMxbqQlz9LxXSg3XZ5PWM0u_cHkHzfydi5_0CsKf0h&3u3840&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=109601",
      "placeUrl": "https://www.google.com/maps/search/?api=1&query=Roxanne's%20Cafe&query_place_id=ChIJg7cNIYyAhYAR8D4ml1oItTg",
      "priceLevel": 2,
      "rating": 4.2,
      "supportingText": "restaurant",
      "userRatingsTotal": 1219,
    },
    Object {
      "coordinates": Array [
        -122.4073542,
        37.7902509,
      ],
      "id": "ChIJaa9OAYyAhYARxwgUHLNEpOY",
      "name": "Bouche",
      "photoUrl": "https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uECKcrbqCoKyZnMn5XOnFD966Cp30QU9X9dOo_T7KeS3n3P0b2K2OsoxMyu8-yJbPBSanZVr1aNknFhWD9IUWfSjlHXWttBocnuB1ntZZV8ANEq5YkZnA-rqRdfN-IIzZqpiRYlQPqkEzcVb5f8F1L8sb1UbX20jMcIu_ykyiGF8rFQZ&3u3264&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=51444",
      "placeUrl": "https://www.google.com/maps/search/?api=1&query=Bouche&query_place_id=ChIJaa9OAYyAhYARxwgUHLNEpOY",
      "priceLevel": 3,
      "rating": 4.5,
      "supportingText": "bar",
      "userRatingsTotal": 465,
    },
    Object {
      "coordinates": Array [
        -122.409088,
        37.7902449,
      ],
      "id": "ChIJvXXIOoyAhYARe-ixq3WdmrY",
      "name": "Sons & Daughters",
      "photoUrl": "https://maps.googleapis.com/maps/api/place/js/PhotoService.GetPhoto?1sAap_uEAbOQ0teYS553dCZEA4txE2wQ53voHdUlQ3-fn5cIed26LOIVIJPWSC7tN5Ie3vhuct5Dw91rpYrP5b8qdw941fIatGOuudqCaA0Ox-MlvCb6mkyVTN5gewLdZsXw5h_QskeQCcltf6-aHXYx2Oa0O-EISmb0NDVQlYqOJmgG-CCMMz&3u1488&5m1&2e1&callback=none&key=AIzaSyDg0Zd8r5Jf5i4MWlM9_eGYrpyRUeqBbPY&token=91263",
      "placeUrl": "https://www.google.com/maps/search/?api=1&query=Sons%20%26%20Daughters&query_place_id=ChIJvXXIOoyAhYARe-ixq3WdmrY",
      "priceLevel": 4,
      "rating": 4.6,
      "supportingText": "restaurant",
      "userRatingsTotal": 249,
    },
  ],
  "selectedFilterOption": null,
}
`;

exports[`Restaurants reducer SET_FILTER_OPTION works as expected 1`] = `
Object {
  "favoritedItemIds": Array [],
  "highlightedRestaurantId": null,
  "isLoading": false,
  "isUpdatingFavorites": false,
  "restaurants": null,
  "selectedFilterOption": "high_to_low",
}
`;

exports[`Restaurants reducer SET_HIGHLIGHTED_RESTAURANT_ID works as expected 1`] = `
Object {
  "favoritedItemIds": Array [],
  "highlightedRestaurantId": "some_id",
  "isLoading": false,
  "isUpdatingFavorites": false,
  "restaurants": null,
  "selectedFilterOption": null,
}
`;

exports[`Restaurants reducer UPDATE_FAVORITED_RESTAURANTS.ERROR works as expected 1`] = `
Object {
  "favoritedItemIds": Array [],
  "highlightedRestaurantId": null,
  "isLoading": true,
  "isUpdatingFavorites": false,
  "restaurants": null,
  "selectedFilterOption": null,
}
`;

exports[`Restaurants reducer UPDATE_FAVORITED_RESTAURANTS.PENDING works as expected 1`] = `
Object {
  "favoritedItemIds": Array [],
  "highlightedRestaurantId": null,
  "isLoading": false,
  "isUpdatingFavorites": true,
  "restaurants": null,
  "selectedFilterOption": null,
}
`;

exports[`Restaurants reducer UPDATE_FAVORITED_RESTAURANTS.SUCCESS works as expected 1`] = `
Object {
  "favoritedItemIds": Array [
    "some_id_1",
    "some_id_2",
  ],
  "highlightedRestaurantId": null,
  "isLoading": false,
  "isUpdatingFavorites": false,
  "restaurants": null,
  "selectedFilterOption": null,
}
`;

exports[`Restaurants reducer UPDATE_FAVORITED_RESTAURANTS.SUCCESS works as expected when empty array 1`] = `
Object {
  "favoritedItemIds": Array [],
  "highlightedRestaurantId": null,
  "isLoading": false,
  "isUpdatingFavorites": false,
  "restaurants": null,
  "selectedFilterOption": null,
}
`;

exports[`Restaurants reducer returns the initial state 1`] = `
Object {
  "favoritedItemIds": Array [],
  "highlightedRestaurantId": null,
  "isLoading": false,
  "isUpdatingFavorites": false,
  "restaurants": null,
  "selectedFilterOption": null,
}
`;
